lib/llvm-backend/src/backend.rs

```diff
@@ -285,13 +298,109 @@ impl LLVMBackend {
 
         let buffer = Arc::new(Buffer::LlvmMemory(memory_buffer));
 
-        (
-            Self {
-                module,
-                buffer: Arc::clone(&buffer),
-            },
-            LLVMCache { buffer },
-        )
+        let raw_stackmap = unsafe {
+            ::std::slice::from_raw_parts(
+                llvm_backend_get_stack_map_ptr(module),
+                llvm_backend_get_stack_map_size(module),
+            )
+        };
+        if raw_stackmap.len() > 0 {
+            let map = stackmap::StackMap::parse(raw_stackmap).unwrap();
+
+            let (code_ptr, code_size) = unsafe {
+                (
+                    llvm_backend_get_code_ptr(module),
+                    llvm_backend_get_code_size(module),
+                )
+            };
+            let mut msm = ModuleStateMap {
+                local_functions: Default::default(),
+                total_size: code_size,
+            };
+
+            let mut local_func_id_to_addr: Vec<usize> = Vec::new();
+
+            // All local functions.
+            for index in module_info.imported_functions.len()..module_info.func_assoc.len() {
+                let name = if cfg!(target_os = "macos") {
+                    format!("_fn{}", index)
+                } else {
+                    format!("fn{}", index)
+                };
+
+                let c_str = CString::new(name).unwrap();
+                let ptr = unsafe { get_func_symbol(module, c_str.as_ptr()) };
```

Comment:
There are other functions (trampolines, etc.) than the ones purely translated from WASM functions. To avoid depending too much on code generation internals I think using the names is a better approach for now.